umh-core/integration/redpanda_extended_test.go

```diff
@@ -0,0 +1,155 @@
+			PrintLogsAndStopContainer()
+		})
+
+		It("should produce messages at high frequency and maintain system stability", func() {
+			By("Configuring redpanda and benthos producers")
+			// Set up redpanda with benthos producers
+			builder := NewRedpandaBuilder()
+			builder.AddGoldenRedpanda()
+			testTopic := "test-throughput"
+
+			for i := 0; i < 10; i++ {
+				builder.AddBenthosProducer(fmt.Sprintf("benthos-%d", i), fmt.Sprintf("%ds", 1+i%3), testTopic)
+			}
+
+			// Apply configuration
+			cfg := builder.BuildYAML()
+			Expect(writeConfigFile(cfg, getContainerName())).To(Succeed())
+
+			By("Waiting for services to stabilize")
+			// Allow time for redpanda and benthos services to start
+			time.Sleep(30 * time.Second)
+
+			By("Monitoring system health during test run")
+			startTime := time.Now()
+			testEndTime := startTime.Add(testDuration)
+
+			// Monitor the health of the system at regular intervals
+			errorsInARow := 0
+			for time.Now().Before(testEndTime) {
+				// Print status update
+				elapsedDuration := time.Since(startTime).Round(time.Second)
+				remainingDuration := time.Until(testEndTime).Round(time.Second)
+
+				GinkgoWriter.Printf(
+					"\n=== Redpanda-Benthos Throughput Test Status ===\n"+
+						"Elapsed time: %v\n"+
+						"Remaining time: %v\n",
+					elapsedDuration, remainingDuration)
+
+				// Check system health
+				err := reportOnMetricsHealthIssue()
+				if err != nil {
+					errorsInARow++
+					GinkgoWriter.Printf("[%d] Error: %v\n", errorsInARow, err)
+				} else {
+					errorsInARow = 0
+				}
+				if errorsInARow >= maxErrorsInARow {
+					Fail("System is unstable, too many errors in a row")
+				}
+
+				time.Sleep(monitorHealthInterval)
+			}
+
+			By("Verifying message count with rpk")
+			// First check if the topic exists
+			out, err := runDockerCommand("exec", getContainerName(), "/opt/redpanda/bin/rpk", "topic", "list")
+			Expect(err).NotTo(HaveOccurred(), "Should be able to list topics with rpk")
+			GinkgoWriter.Printf("Available topics:\n%s\n", out)
+
+			// Count the messages by using rpk to consume all messages and pipe through wc -l
+			out, err = runDockerCommand("exec", getContainerName(), "bash", "-c",
+				fmt.Sprintf("/opt/redpanda/bin/rpk topic consume %s --offset earliest -n -1 2>/dev/null | wc -l", testTopic))
+			Expect(err).NotTo(HaveOccurred(), "Should be able to count messages with rpk")
+
+			// Parse message count
+			messageCount := 0
+			_, err = fmt.Sscanf(strings.TrimSpace(out), "%d", &messageCount)
+			Expect(err).NotTo(HaveOccurred(), "Should be able to parse message count")
+
+			// Calculate expected message count with a tolerance of 20% loss
+			totalSeconds := int(testDuration.Seconds())
+			expectedMessagesPerSecond := 10 * messagesPerSecond // 10 producers * 10 messages per second
+			expectedMessages := totalSeconds * expectedMessagesPerSecond
+			minimumExpectedMessages := int(float64(expectedMessages) * 0.8) // Allow for 20% message loss
```

Comment:
20% seems to be a lot. we can start with this to develop the test, but ideally we land up at 95%. maybe need to have a grace period for the first 20 seconds or so to allow all services starting up.

Reply:
Updated it to 95%, which seems reachable.